packages/manager/modules/billing/src/components/commitment/Commitment.class.js

```diff
@@ -0,0 +1,75 @@
+import { Pricing } from '@ovh-ux/manager-models';
+
+export default class Commitment {
+  constructor(pricing) {
+    Object.assign(this, {
+      pricing: new Pricing(pricing),
+      configuration: pricing.engagement_configuration,
+      pricingMode: pricing.pricingMode,
+    });
+
+    this.price = this.pricing.price.value;
+  }
+
+  addOptionCommitment(optionEngagement) {
+    this.price += optionEngagement.price;
+    this.pricing = new Pricing({
+      ...this.pricing,
+      price: {
+        ...this.pricing.price,
+        value: this.price,
+      },
+    });
+  }
+
+  get durationInMonths() {
+    return (
+      moment
+        .duration(this.pricing.interval, this.pricing.intervalUnit)
+        .asMonths() *
+      moment
+        .duration(this.configuration.duration, this.configuration.duration_unit)
+        .asMonths()
+    );
+  }
+
+  get totalPrice() {
+    return new Pricing({
+      interval: this.durationInMonths,
+      intervalUnit: 'month',
+      price: {
+        ...this.pricing.price,
+        value: this.durationInMonths * this.pricing.monthlyPriceValue,
+      },
+    }).format();
+  }
+
+  getPriceDiff(commitment) {
+    return new Pricing({
+      interval: this.durationInMonths,
+      intervalUnit: 'month',
+      price: {
+        ...this.pricing.price,
+        value: this.totalPrice.value - commitment.totalPrice.value,
+      },
+    }).format();
+  }
+
+  get commitmentType() {
+    return this.configuration.type;
+  }
+
+  isPeriodic() {
+    return this.commitmentType === 'periodic';
```

Comment:
'periodic' and 'upfront' can be defined in constants file

Reply:
They could but isn't it redundant as we created the class to abstract those values and use those values just through the `isUpfront` and `isPeriodic` functions